tensorflow/core/kernels/adjust_hue_op.cc

```diff
@@ -185,56 +136,100 @@ static void hv_range_to_rgb(float h, float v_min, float v_max, float* r,
 }
 }  // namespace internal
 
-template <>
-class AdjustHueOp<CPUDevice> : public AdjustHueOpBase {
+template<typename Device>
+class AdjustHueOp : public OpKernel {
```

Comment:
@zheng-xq Somehow I thought that you didn't want the subclassing, in addition to suggesting the functor pattern to address the GPU compilation units. If it's OK to leave for now, I can fix that when I do the saturation PR later. Hopefully it's not too much template bloat relative to Eigen. :)